src/main/scala/model/Bunny.scala

```diff
@@ -0,0 +1,142 @@
+      .reduce(_ + "\n" + _)
+      .replace("_", " ") + "\n"
+  }
+}
+
+/**
+ * Represents a Bunny which is still alive.
+ * @param genotype  contains the Bunny genes, it must be completed (contain all the available Genes of the world)
+ * @param mom       the bunny's mom
+ * @param dad       the bunny's dad
+ * @param age       how many generations the Bunny has spent in the world.
+ */
+case class AliveBunny(genotype: CompletedGenotype, mom:Option[Bunny], dad:Option[Bunny], age:Int) extends Bunny{
+  override val isAlive: Boolean = true
+}
+
+/**
+ * Represents a Bunny which dead.
+ * @param genotype  contains the Bunny genes, it must be completed (contain all the available Genes of the world)
+ * @param mom       the bunny's mom
+ * @param dad       the bunny's dad
+ * @param age       how many generations the Bunny has spent in the world.
+ */
+case class DeadBunny(genotype: CompletedGenotype, mom:Option[Bunny], dad:Option[Bunny], age:Int) extends Bunny{
+  override val isAlive: Boolean = false
+}
+
+/**
+ * Represents the abstraction of a Bunny that as just been created.
+ */
+abstract class InitBunny() extends Bunny{
+  override val age: Int = 0
+  override val isAlive: Boolean = true
+}
+
+/**
+ * Represents a Bunny that as just been created.
+ */
+case class ChildBunny(genotype: CompletedGenotype, mom:Option[Bunny], dad:Option[Bunny]) extends InitBunny
+
+/**
+ * Represents the first Bunny which appears in the world, so it does not have a mom and a dad.
+ */
+case class FirstBunny(genotype: CompletedGenotype) extends InitBunny {
+  override val mom: Option[Bunny] = Option.empty
+  override val dad: Option[Bunny] = Option.empty
+}
+
+object BunnyUtils {
+  val CHILDREN_NUMBER = 4
+
+  /**
+   * @return a FirstBunny with the "base" allele for each gene
+   */
+  def getBaseFirstBunny: FirstBunny =
+    FirstBunny(CompletedGenotype(Genes.values.unsorted.map(gk =>
+      (gk, StandardGene(gk, StandardAllele(gk.base), StandardAllele(gk.base)))).toMap))
+
+  /**
+   * @return a FirstBunny with a random allele for each gene
+   */
+  def getRandomFirstBunny: FirstBunny = {
+    FirstBunny(
+      CompletedGenotype(
+        Genes.values.unsorted.map(gk => {
+          (gk, StandardGene(gk, StandardAllele(List(gk.base, gk.mutated)(Random.nextInt(2))),
+                                StandardAllele(List(gk.base, gk.mutated)(Random.nextInt(2)))))
+        }).toMap
+      )
+    )
+  }
+
+  /**
+   * @param bunnies   a seq of bunnies
+   * @return          a seq of random couples formed from all of the bunnies (or most of them, if they are odd)
+   */
+  def getCouples(bunnies: Seq[Bunny]): Seq[(Bunny, Bunny)] = {
+    val split = Random.shuffle(bunnies).splitAt(bunnies.size/2)
+    split._1.zip(split._2)
+  }
+
+  /**
+   * @param mom a bunny
+   * @param dad another bunny
+   * @return    the 4 children of the couple, one for each cell of the Punnett's square
+   */
+  def getChildren(mom: Bunny, dad:Bunny): Seq[Bunny] = {
```

Comment:
forse il nome "getChildren" è un po' fuorviante siccome è una factory, stessa cosa per "getAllChildren"

Reply:
Giusto, "generateChildren" potrebbe andare? Posso modificare anche gli altri get in questo modo (ex. getBaseFirstBunny)

Reply:
Per me va benissimo

Reply:
Per me anche va bene